File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Querying SQL Azure Data from Web Parts/C#/AZURE_QueryingWebPart/AZURE_QueryingWebPart/AZURE_QueryingWebPartUserControl.ascx.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;

namespace AZURE_QueryingWebPart.AZURE_QueryingWebPart
{
    public partial class AZURE_QueryingWebPartUserControl : UserControl
    {
        //This is the query to run. 
        const string QUERY_STRING = "SELECT * FROM Products.dbo.Products;";
        //This is the connection string. Make sure the servername include the tcp
        //protocol, and the name of your SQL Azure server, including ".database.windows.net". 
        //Edit the username and password.
        //Also note the longer connection timeout.
        const string CONNECTION_STRING = "Server=tcp:myservername.database.windows.net;Database=Products;User ID=Administrator;Password=PASSWORD;Trusted_Connection=False;Encrypt=True;Connection Timeout=30";

        private DataSet productsDataset = new DataSet();

        protected void Page_Load(object sender, EventArgs e)
        {
        }

        protected void lnkbtnGetProducts_Click(object sender, EventArgs e)
        {
            //Create a connection to SQL Azure
            using (SqlConnection sqlConnection = new SqlConnection(CONNECTION_STRING))
            {
                SqlDataAdapter dataAdapter = new SqlDataAdapter();
                //Specify the select query
                dataAdapter.SelectCommand = new SqlCommand(QUERY_STRING, sqlConnection);
                //Get the data
                dataAdapter.Fill(productsDataset);
                //Bind the data adaptor to the datagrid control
                datagridProducts.DataSource = productsDataset;
                datagridProducts.DataBind();
            }
        }
    }
}

File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/[VB]-SharePoint 2010 101 Code Samples/SharePoint 2010 Querying SQL Azure Data from Web Parts/VB/AZURE_QueryingWebPart/AZURE_QueryingWebPart/AZURE_QueryingWebPartUserControl.ascx.vb
Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts

Partial Public Class AZURE_QueryingWebPartUserControl
    Inherits UserControl

    'This is the query to run. 
    Const QUERY_STRING As String = "SELECT * FROM Products.dbo.Products;"
    'This is the connection string. Make sure the servername include the tcp
    'protocol, and the name of your SQL Azure server, including ".database.windows.net". 
    'Edit the username and password.
    'Also note the longer connection timeout.
    Const CONNECTION_STRING As String = "Server=tcp:myservername.database.windows.net;Database=Products;User ID=administrator;Password=password;Trusted_Connection=False;Encrypt=True;Connection Timeout=30"

    Private productsDataset As DataSet = New DataSet()

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load

    End Sub

    Protected Sub lnkbtnGetProducts_Click(ByVal sender As Object, ByVal e As EventArgs) Handles lnkbtnGetStoreInformation.Click
        'Create a connection to SQL Azure
        Using sqlConnection As SqlConnection = New SqlConnection(CONNECTION_STRING)
            Dim dataAdapter As SqlDataAdapter = New SqlDataAdapter()
            'Specify the select query
            dataAdapter.SelectCommand = New SqlCommand(QUERY_STRING, sqlConnection)
            'Get the data
            dataAdapter.Fill(productsDataset)
            'Bind the data adaptor to the datagrid control
            datagridProducts.DataSource = productsDataset
            datagridProducts.DataBind()
        End Using
    End Sub
End Class
